src/pythonic/test_primes.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+import itertools
+from main import Primes, Sieve
+
+
+def test_sieve_limit():
+    limit = 10000
+    with Sieve(limit) as s:
+        assert s.upper_bound() >= limit
+
+
+def test_upper_bound_exception():
```

Comment:
Could this be renamed to something like `test_checking_above_limit_is_an_error`? I think `upper_bound_exception` is a little bit too vague.